addon/components/tf-button/component.js

```diff
@@ -0,0 +1,35 @@
+/**
+  @module ticketfly-buttons
+ */
+import Ember from 'ember';
+import layout from './template';
+import { collect } from 'ember-computed';
+const { Component, computed, get } = Ember;
+
+/**
+  @public
+  @class TfButton
+  @extends Ember.Component
+ */
+export default Component.extend({
+  layout,
+
+  tagName: 'button',
+  classNames: ['c-button'],
+  attributeBindings: ['disabled'],
+  classNameBindings: ['buttonCategory'],
+
+  utilityGroups: collect('weightGroup', 'colorGroup'),
+
+  buttonCategory: computed('utilityGroups', {
+    get() {
+      let buttonCategory = '';
+      get(this, 'utilityGroups').map((group) => {
+        if (group) {
+          buttonCategory += `u-button-${group} `;
```

Comment:
As we discussed yesterday, I think that rather than using what exactly what was in the original issue spec, we should use the BEM-like class names for the various states. 

So, I guess that'd mean `classNames: ['ticketfly-button']` and `u-button-${group}` would become `ticketfly-button--${group}`.